### PR TITLE
Extract vuex implementation for consistency and testability

### PR DESCRIPTION
**Pull request by guergana**
Bug: [T295704](https://phabricator.wikimedia.org/T295704)

### CODE REVIEW

File: resources/js/store/index.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import Vue from 'vue';
+import Vuex, {Store} from 'vuex';
+import { Inertia } from '@inertiajs/inertia';
+import mutations from './mutations';
+
+export function getInitialState() {
```
Comment (itamargiv) [inline resources/js/store/index.ts:6]: Nice one, we can use RootState in here as well.

File: resources/js/store/index.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    return {
+        loading: false,
+        lastSearchedIds: ''
+    }
+}
+
+export function createStore(): Store<{loading: boolean}>{
```
Comment (itamargiv) [inline resources/js/store/index.ts:13]:
```suggestion
export function createStore(): Store<RootState>{
```

**Review by itamargiv — COMMENTED** [review]
Great job, thank you! One small comment but non blocker

File: resources/js/store/index.ts (hunk 2 above)
Comment (guergana) [reply resources/js/store/index.ts:13]: oh weird. i dont remember adding that. maybe it was automatically added.

File: resources/js/store/index.ts (hunk 1 above)
Comment (guergana) [reply resources/js/store/index.ts:6]: But RootState is just the types. getInitialState returns the initial values. I think they are different things.

Comment (itamargiv) [reply resources/js/store/index.ts:7]: Yes, I mean that we should annotate the return type of this function with RootState, that's what it returns after all. No?

Comment (guergana) [reply resources/js/store/index.ts:7]: yes! Excuse moi, I misunderstood your comment :)

Comment (guergana) [reply resources/js/store/index.ts]: Annotated!

**Review by itamargiv — APPROVED** [review]
Thank yout, looks great!

File: resources/js/store/index.ts (hunk 1 above)
Comment (itamargiv) [reply resources/js/store/index.ts]: np, probably wasn't clear enough. Thx!

**Pull request merged**